paimon-format/src/main/java/org/apache/paimon/format/parquet/writer/RowDataParquetBuilder.java

```diff
@@ -77,14 +75,10 @@ public ParquetWriter<InternalRow> createWriter(OutputFile out, String compressio
                 .build();
     }
 
-    public String getCompression(@Nullable String compression) {
-        String compressName;
-        if (null != compression) {
-            compressName = compression;
-        } else {
-            compressName =
-                    conf.get(ParquetOutputFormat.COMPRESSION, CompressionCodecName.SNAPPY.name());
+    public String getCompression(String compression) {
+        if (conf.get("parquet.compression") != null) {
+            return conf.get("parquet.compression");
         }
-        return compressName;
+        return compression;
```

Comment:
return conf.getString("parquet.compression", compression)?

Reply:
you are right... updated